configdialog.cpp

```diff
@@ -544,66 +544,259 @@ std::wstring PaletteCustomizeDialog::export_palettes() {
 		auto *data = palette.data;
 
 		palettes_string.append(palette.name + L": ");
-		palettes_string.append(m_palette_repo.serialize(*data));
-		palettes_string.append(L";\r\n");
+		std::wstring colors = m_palette_repo.serialize(*data) + L";";
+		// Add extra "fake" colors so that a palette definition is guaranteed to be longer than the maximum allowed name length
+		while (colors.length() < PaletteCustomizeDialog::MaxPaletteNameSize) {
+			colors.append(L"#000000;");
+		}
+		palettes_string.append(colors);
+		palettes_string.append(L"\r\n");
 	}
 	palettes_string.append(L"~");
 
 	return palettes_string;
 }
 
 void PaletteCustomizeDialog::import_palettes(std::wstring *palettes) {
-	static std::wstring valid_chars = L"qwertyuiopasdfghjklzxcvbnmQWERTYUIOPASDFGHJKLZXCVBNM0123456789-_!?:";
-	static std::wstring valid_hex_chars = L"0123456789abcdefABCDEF";
-	size_t current_index = palettes->find_first_of(valid_chars.c_str());
-	std::wstring last_successful_palette_name;
-	std::wstring current_name;
-	std::wstring current_data[7];
+	using namespace std;
```

Comment:
evil.

Reply:
This is a sign you have gone too far off the deep end. If writing `std::` is just getting too tedious and making things too ugly, you are probably doing too much stuff.

The main reason I don't `using namespace` except for truly deranged namespaces like `chrono` or `string_literals` is because `using namespace` destroys all of the safety provided by namespaces. The `std` namespace contains A LOT of crap, much of it with really simple names, like `begin` or `get`. If I write a function name or class name that accidentally collides with something in that namespace, what the hell happens? I don't know and I don't want to find out. 

That being said, you did do the right thing by containing the `using` to just this function.